File: test/fixtures/sample.ts
import { makeToken } from "./auth";

interface Service {
  ping(): string;
}

class BaseService implements Service {
  ping(): string {
    return "ok";
  }
}

export class UserService extends BaseService {
  login(user: string): string {
    const token = makeToken(user);
    return this.store(token);
  }

  store(token: string): string {
    return token;
  }
}

export function bootstrap(): string {
  const svc = new UserService();
  return svc.login("alice");
}

File: test/fixtures/sample.py
from auth.tokens import make_token


class BaseService:
    def ping(self):
        return "ok"


class UserService(BaseService):
    def login(self, user):
        token = make_token(user)
        return self.store(token)

    def store(self, token):
        return token


def bootstrap():
    svc = UserService()
    return svc.login("alice")
